tests/unit_tests/trading_tests/test_exchange_market_status_fixer.py

```diff
@@ -0,0 +1,239 @@
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+from math import nan
+
+from config import ExchangeConstantsMarketStatusColumns as Ecmsc
+from config import ExchangeConstantsMarketStatusInfoColumns as Ecmsic
+from trading import ExchangeMarketStatusFixer
+
+
+class TestExchangeMarketStatusFixer:
+
+    @staticmethod
+    def _get_limits(amount_min=None, amount_max=None,
+                    price_min=None, price_max=None,
+                    cost_min=None, cost_max=None):
+        return {
+            Ecmsc.LIMITS_AMOUNT.value: {
+                Ecmsc.LIMITS_AMOUNT_MIN.value: amount_min,
+                Ecmsc.LIMITS_AMOUNT_MAX.value: amount_max,
+            },
+            Ecmsc.LIMITS_PRICE.value: {
+                Ecmsc.LIMITS_PRICE_MIN.value: price_min,
+                Ecmsc.LIMITS_PRICE_MAX.value: price_max,
+            },
+            Ecmsc.LIMITS_COST.value: {
+                Ecmsc.LIMITS_COST_MIN.value: cost_min,
+                Ecmsc.LIMITS_COST_MAX.value: cost_max,
+            },
+        }
+
+    @staticmethod
+    def _get_precision(precision_amount=None, precision_cost=None, precision_price=None):
+        return {
+            Ecmsc.PRECISION_AMOUNT.value: precision_amount,
+            Ecmsc.PRECISION_COST.value: precision_cost,
+            Ecmsc.PRECISION_PRICE.value: precision_price,
+        }
+
+    # Global
+    def test_exchange_market_status_fixer_without_precision_cost_and_amount_with_price(self):
+        current_price = 1564.1458
+        ms = {
+            Ecmsc.PRECISION.value: self._get_precision(nan, None, nan),
+            Ecmsc.LIMITS.value: self._get_limits(None, nan, 0.05, 1e4, nan, None)
+        }
+
+        assert ExchangeMarketStatusFixer(ms, price_example=current_price).get_market_status() == {
+            Ecmsc.PRECISION.value: self._get_precision(4, 4, 4),
+            Ecmsc.LIMITS.value: self._get_limits(0.007421331729698568, 344.46770487105107,
+                                                 1.5641458, 1564145.8,
+                                                 1.5641458, 1564145.8)
+        }
+
+        current_price = 1.56e-6
+        ms = {
+            Ecmsc.PRECISION.value: self._get_precision(nan, None, nan),
+            Ecmsc.LIMITS.value: self._get_limits(None, nan, 0, 0, nan, None)
+        }
+
+        assert ExchangeMarketStatusFixer(ms, price_example=current_price).get_market_status() == {
+            Ecmsc.PRECISION.value: self._get_precision(8, 8, 8),
+            Ecmsc.LIMITS.value: self._get_limits(513231262.1957913, 345077731.54405594,
```

Comment:
513231262.1957913 is min amount for something with a price of 1.56e-6  ? => if price in BTC, min amount is 800.64 BTC ?

Reply:
Fixed !